astropy/coordinates/solar_system.py

```diff
@@ -0,0 +1,276 @@
+        or a valid integer index.
+
+    earth_time : `~astropy.time.Time`
+        Time used for position of Earth. When correcting for light travel time,
+        one wants to use different times for the body in question and Earth.
+        If this is set to ```None```, the same time is used for both.
+
+    Returns
+    -------
+    earth_body_vector : `~astropy.coordinates.CartesianRepresentation`
+        Barycentric (ICRS) vector from Geocenter to the body in cartesian coordinates
+
+    earth_distance : `~astropy.units.Quantity`
+        Distance between Earth and body.
+
+    Notes
+    -----
+
+    """
+    earth_time = earth_time if earth_time is not None else time
+    earth_loc = get_barycentric_body_position(earth_time, 'earth')
+    body_loc = get_barycentric_body_position(time, body_key)
+
+    earth_body_vector = body_loc.xyz - earth_loc.xyz
+
+    earth_distance = np.sqrt(np.sum(earth_body_vector**2, axis=0))
+    return earth_body_vector, earth_distance
+
+
+def _get_apparent_body_position(time, body_key):
+    """
+    Calculate the apparent position of body ``body_key`` in cartesian
+    coordinates, given the approximate light travel time to the object.
+
+    Uses ``jplephem`` with the DE430 kernel.
+
+    Parameters
+    ----------
+    time : `~astropy.time.Time`
+        Time of observation
+
+    body_key : int, str
+        The solar system body to calculate. Can be a string, e.g. 'moon',
+        or a valid integer index.
+
+    Returns
+    -------
+    cartesian_position : `~astropy.coordinates.CartesianRepresentation`
+        Barycentric (ICRS) apparent position of the body in cartesian coordinates
+    """
+    # Get distance of body at `time`
+    earth_to_body_vector, earth_distance = _get_earth_body_vector(time, body_key)
+
+    # The apparent position depends on the time that the light was emitted from
+    # the distant body, so subtract off the light travel time
+    light_travel_time = earth_distance/speed_of_light
+    emitted_time = time - light_travel_time
+
+    # Calculate position given approximate light travel time.
+    delta_light_travel_time = 20*u.s
+    while (np.fabs(delta_light_travel_time)) > 1.0e-8*u.s:
+        earth_to_body_vector, earth_distance = _get_earth_body_vector(emitted_time, body_key)
+        delta_light_travel_time = light_travel_time - earth_distance/speed_of_light
+        light_travel_time = earth_distance/speed_of_light
+        emitted_time = time - light_travel_time
+
+    return get_barycentric_body_position(emitted_time, body_key)
+
+
+def get_body(time, body_key, location=None):
+    """
+    Get a `~astropy.coordinates.SkyCoord` for a body as observed from a
+    location on Earth.
+
+    Parameters
+    ----------
+    time : `~astropy.time.Time`
+        Time of observation
+
+    body_key : int, str
+        The solar system body to calculate. Can be a string, e.g. 'moon',
+        or a valid integer index.
```

Comment:
A nice way to do this is to show *how* to get the defined names.  In `Time` (https://github.com/astropy/astropy/blob/master/astropy/time/core.py#L127), @taldcroft wrote this neatly as a doctest, so it immediately checks that the list is correct.  For this purpose, it is handy to use an `OrderedDict`...